### PR TITLE
[CHK-875] update all classes to match commons library

### PR DESCRIPTION
**Pull request by ciuffagianluca**
#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateResultHandler.java [hunk 1]
```diff
@@ -111,20 +113,36 @@ public Mono<TransactionActivatedEvent> handle(TransactionActivateResultCommand c
...
+                        null);
+
 
               TransactionActivatedEvent transactionActivatedEvent =
                   new TransactionActivatedEvent(
                       transactionId,
-                      command.getData().transactionActivationRequested().getRptId().value(),
-                      saved.paymentToken(),
+                      transactionActivationRequested.getNoticeCodes().stream()
+                          .map(noticeCode2 -> new it.pagopa.ecommerce.commons.documents.NoticeCode(
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateResultHandler.java:140]: same as above

File: src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateResultHandler.java [hunk 2]
```diff
@@ -139,15 +157,15 @@ public Mono<TransactionActivatedEvent> handle(TransactionActivateResultCommand c
                         .doOnError(
                                 exception ->
                                     log.error(
-                                            "Error to generate event TRANSACTION_ACTIVATED_EVENT for rptId {} and transactionId {} - error {}",
-                                            transactionActivatedEvent.getRptId(),
+                                            "Error to generate event TRANSACTION_ACTIVATED_EVENT for rptIds {} and transactionId {} - error {}",
+                                            String.join(",",transactionActivatedEvent.getNoticeCodes().stream().map(noticeCode1 -> noticeCode1.getRptId()).toList()),
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateResultHandler.java:161]: why the odd name noticeCode1? would be better noticeCode?

File: src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateResultHandler.java [hunk 3]
```diff
@@ -139,15 +157,15 @@ public Mono<TransactionActivatedEvent> handle(TransactionActivateResultCommand c
...
                                             transactionActivatedEvent.getTransactionId(),
                                             exception.getMessage()))
                         .doOnNext(
                                 event ->
                                         log.info(
-                                                "Generated event TRANSACTION_ACTIVATED_EVENT for rptId {} and transactionId {}",
-                                                event.getRptId(),
+                                                "Generated event TRANSACTION_ACTIVATED_EVENT for rptIds {} and transactionId {}",
+                                                String.join(",", event.getNoticeCodes().stream().map(noticeCode1 -> noticeCode1.getRptId()).toList()),
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateResultHandler.java:168]: same as above

File: src/main/java/it/pagopa/transactions/commands/handlers/TransactionAddUserReceiptHandler.java [hunk 4]
```diff
@@ -75,8 +73,15 @@ public Mono<TransactionUserReceiptAddedEvent> handle(TransactionAddUserReceiptCo
 
                     TransactionUserReceiptAddedEvent event = new TransactionUserReceiptAddedEvent(
                             command.getData().transaction().getTransactionId().value().toString(),
-                            command.getData().transaction().getRptId().value(),
-                            command.getData().transaction().getTransactionActivatedData().getPaymentToken(),
+                            command.getData().transaction().getNoticeCodes().stream().map(noticeCode ->
+                                    new NoticeCode(
+                                            command.getData().transaction().getTransactionActivatedData().getNoticeCodes().
+                                    stream().filter(noticeCode1 -> noticeCode1.getRptId().equals(noticeCode.rptId().value())).findFirst().get()
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/commands/handlers/TransactionAddUserReceiptHandler.java:79]: why not just noticeCode.getPaymentToken()?

File: src/main/java/it/pagopa/transactions/commands/handlers/TransactionRequestAuthorizationHandler.java [hunk 5]
```diff
@@ -65,12 +66,20 @@ public Mono<RequestAuthorizationResponseDto> handle(TransactionRequestAuthorizat
...
+
                                 TransactionAuthorizationRequestedEvent authorizationEvent = new TransactionAuthorizationRequestedEvent(
                                         transaction.getTransactionId().value().toString(),
-                                        transaction.getRptId().value(),
-                                        transaction.getTransactionActivatedData().getPaymentToken(),
+                                        transaction.getNoticeCodes().stream().map(
+                                                noticeCode ->  new NoticeCode(
+                                                        transaction.getTransactionActivatedData().getNoticeCodes().
+                                                                stream().filter(noticeCode1 -> noticeCode1.getRptId().equals(noticeCode.rptId().value())).findFirst().get()
+                                                                .getPaymentToken(),
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/commands/handlers/TransactionRequestAuthorizationHandler.java:76]: same as in src/main/java/it/pagopa/transactions/commands/handlers/TransactionAddUserReceiptHandler.java

File: src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java [hunk 6]
```diff
@@ -88,8 +89,15 @@ public Mono<TransactionClosureSentEvent> handle(TransactionClosureSendCommand co
 
                                 TransactionClosureSentEvent event = new TransactionClosureSentEvent(
                                         command.getData().transaction().getTransactionId().value().toString(),
-                                        command.getData().transaction().getRptId().value(),
-                                        command.getData().transaction().getTransactionActivatedData().getPaymentToken(),
+                                        command.getData().transaction().getNoticeCodes().stream().map(
+                                                noticeCode ->  new NoticeCode(
+                                                        command.getData().transaction().getTransactionActivatedData().getNoticeCodes().
+                                                                stream().filter(noticeCode1 -> noticeCode1.getRptId().equals(noticeCode.rptId().value())).findFirst().get()
+                                                                .getPaymentToken(),
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java:96]: same as src/main/java/it/pagopa/transactions/commands/handlers/TransactionAddUserReceiptHandler.java

File: src/main/java/it/pagopa/transactions/projections/handlers/TransactionsActivationProjectionHandler.java [hunk 7]
```diff
@@ -24,23 +26,26 @@
...
-		PaymentToken paymentToken = new PaymentToken(event.getPaymentToken());
-		RptId rptId = new RptId(event.getRptId());
-		TransactionDescription description = new TransactionDescription(data.getDescription());
-		TransactionAmount amount = new TransactionAmount(data.getAmount());
+		List<NoticeCode> noticeCodeList = event.getNoticeCodes().stream().map(noticeCode ->
+			new NoticeCode(
+					new PaymentToken(noticeCode.getPaymentToken()),
+					new RptId(noticeCode.getRptId()),
+					new TransactionAmount(data.getNoticeCodes().stream().filter(noticeCode1 -> noticeCode1.getRptId().equals(noticeCode.getRptId())).findFirst().get().getAmount()),
+					new TransactionDescription(data.getNoticeCodes().stream().filter(noticeCode1 -> noticeCode1.getRptId().equals(noticeCode.getRptId())).findFirst().get().getDescription()))
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/projections/handlers/TransactionsActivationProjectionHandler.java:34]: why not just noticeCode.getAmount() and noticeCode.getDescription()?

**Review by pietro-tota — APPROVED** [review]
🚀 Well done! just some comments

File: src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java [hunk 8]
```diff
@@ -54,14 +54,15 @@ private Mono<PostePayAuthResponseEntityDto> requestPostepayAuthorization(Authori
                 .switchIfEmpty(Mono.empty())
             .map(authorizationRequestData ->
                 new PostePayAuthRequestDto()
-                    .grandTotal(BigDecimal.valueOf(((long) authorizationData.transaction().getAmount().value()) + authorizationData.fee()))
-                    .description(authorizationData.transaction().getDescription().value())
+                    .grandTotal(BigDecimal.valueOf(((long) authorizationData.transaction().getNoticeCodes().stream().mapToInt(noticeCode -> noticeCode.transactionAmount().value()).sum()) + authorizationData.fee()))
+                    .description(authorizationData.transaction().getNoticeCodes().get(0).transactionDescription().value())
```
Comment (infantesimone) [inline src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java:58]: TODO handle multiple description due to multiple notice code

File: src/main/java/it/pagopa/transactions/projections/handlers/TransactionsActivationProjectionHandler.java (hunk 7 above)
Comment (ciuffagianluca) [reply src/main/java/it/pagopa/transactions/projections/handlers/TransactionsActivationProjectionHandler.java:34]: They comes from main branch. This datas come from TransactionEventActivatedData so they come from the same origin.

File: src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java (hunk 8 above)
Comment (ciuffagianluca) [reply src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java:59]: Task opened

**Review by infantesimone — DISMISSED** [review]

File: src/test/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandlerTest.java [hunk 9]
```diff
@@ -225,25 +263,29 @@ void shouldSetTransactionStatusToClosureFailedOnNodoKO() {
...
         TransactionClosureSentEvent event = new TransactionClosureSentEvent(
                 transactionId.toString(),
-                transactionActivatedEvent.getRptId(),
-                transactionActivatedEvent.getData().getPaymentToken(),
+                transactionActivatedEvent.getNoticeCodes().stream().map(noticeCode -> new it.pagopa.ecommerce.commons.documents.NoticeCode(
+                        noticeCode.getRptId(),
+                        transactionActivatedEvent.getData().getNoticeCodes().stream().filter(noticeCode1 -> noticeCode1.getRptId().equals(noticeCode.getRptId())).findFirst().get().getPaymentToken(),
+                        noticeCode.getDescription(),
+                        noticeCode.getAmount()
+                )).toList(),
```
Comment (giovanniberti) [inline src/test/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandlerTest.java:275]:
```suggestion
                transactionActivatedEvent.getNoticeCodes(),
```

Comment (ciuffagianluca) [reply src/test/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandlerTest.java:275]: ok

**Pull request merged**